### PR TITLE
fix(datasource/conan): continue on error during fetching package url

### LINKED ISSUE #23373: Renovate fails to update conan packages without conan.package.url property
### Discussed in https://github.com/renovatebot/renovate/discussions/23369

<div type='discussions-op-text'>

<sup>Originally posted by **jens-kretschmer** July 14, 2023</sup>
### How are you running Renovate?

Self-hosted Renovate

### If you're self-hosting Renovate, tell us what version of Renovate you run.

35.159.7

### If you're self-hosting Renovate, select which platform you are using.

GitLab self-hosted

### Was this something which used to work for you, and then stopped?

It used to work, and then stopped

### Describe the problem

Our self hosted renovate bot recently stopped to update the conan dependencies in all our projects. It generates GitLab issues stating that it cannot look up the dependencies. E.g.

`Renovate failed to look up the following dependencies: Failed to look up conan package private-conan-package/1.2.102@_/_,`

From the debug logs I can see that renovate tries to fetch the conan.package.url property which was recently introduced: https://github.com/renovatebot/renovate/pull/22656

We do not set that property on our conan packages as we don't need it.

I would need a public conan repository with packages without the conan.package.url property to create a minimal reproducer repository.

### Relevant debug logs

<details><summary>Logs</summary>

```
DEBUG: GET https://private-jfrog.com/artifactory/api/storage/conan-repo/_/private-conan-package/1.2.116/_/0/export/conanfile.py?properties=conan.package.url = (code=ERR_NON_2XX_3XX_RESPONSE, statusCode=404 retryCount=0, duration=241) (repository=private-gitlab-repo)
DEBUG: Datasource 404 (repository=private-gitlab-repo)
       "datasource": "conan",
       "packageName": "private-conan-package/1.2.102@_/_",
       "url": "https://private-jfrog.com/artifactory/api/storage/conan-repo/_/private-conan-package/1.2.116/_/0/export/conanfile.py?properties=conan.package.url"
DEBUG: Failed to look up conan package private-conan-package/1.2.102@_/_ (repository=private-gitlab-repo, packageFile=conanfile.txt, dependency=private-conan-package/1.2.102@_/_)
```

</details>

I replaced the URLs, the package names and the gitlab repo name as I'm not allowed to share that information.

### Have you created a minimal reproduction repository?

I have explained in the description why a minimal reproduction is impossible</div>

### PR DESCRIPTION
**Pull request by MartijnLeijssen-TomTom**
## Changes
Ignore errors during fetching package url of a conan package when using jfrog artifactory

## Context
- Fixes #23373

## Documentation (please check one with an [x])

- [ ] I have updated the documentation, or
- [X] No documentation update is required

## How I've tested my work (please select one)

I have verified these changes via:

- [ ] Code inspection only, or
- [X] Newly added/modified unit tests, or
- [ ] No unit tests but ran on a real repository, or
- [ ] Both unit tests + ran on a real repository

### CODE REVIEW

File: lib/modules/datasource/conan/index.ts [hunk 1]
```diff
@@ -153,48 +153,59 @@ export class ConanDatasource extends Datasource {
...
-            );
-            const packageUrlResp = await this.http.getJson<ConanProperties>(
-              packageUrl
-            );
-            const conanPackageUrl =
-              packageUrlResp.body.properties['conan.package.url'][0];
-            dep.sourceUrl = conanPackageUrl;
+          } catch (err) {
+            // eslint-disable-next-line @typescript-eslint/restrict-template-expressions
+            logger.warn(`Couldn't determine conan package url ${err}`);
```
Comment (viceice) [inline lib/modules/datasource/conan/index.ts:208]:
```suggestion
            logger.debug({ err }, `Couldn't determine conan package url`);
```
this error isn't Essential

**Review by viceice — DISMISSED** [review]

File: lib/modules/datasource/conan/index.spec.ts [hunk 2]
```diff
@@ -383,5 +383,94 @@ describe('modules/datasource/conan/index', () => {
         })
       ).toBeNull();
     });
+
```
Comment (HonkingGoose) [inline lib/modules/datasource/conan/index.spec.ts:386]: This PR does not yet automatically close the relevant issue?

Correct syntax is: `Fixes #23373`

**Review by HonkingGoose — CHANGES_REQUESTED** [review]

File: lib/modules/datasource/conan/index.ts [hunk 3]
```diff
@@ -153,48 +153,58 @@ export class ConanDatasource extends Datasource {
...
-              `${user}/${conanPackage.conanName}/${latestVersion}/${channel}/${packageRev}/export/conanfile.py?properties=conan.package.url`
-            );
-            const packageUrlResp = await this.http.getJson<ConanProperties>(
-              packageUrl
-            );
-            const conanPackageUrl =
-              packageUrlResp.body.properties['conan.package.url'][0];
-            dep.sourceUrl = conanPackageUrl;
+          } catch (err) {
+            logger.debug({ err }, "Couldn't determine conan package url");
```
Comment (HonkingGoose) [inline lib/modules/datasource/conan/index.ts:207]:
```suggestion
            logger.debug({ err }, "Couldn't determine Conan package url");
```

I think we should capitalize the proper noun Conan here?

**Review by HonkingGoose — CHANGES_REQUESTED** [review]
The PR will now close the correct issue.

Maybe one small style fix and we're good to go, from my end, at least.

**Review by HonkingGoose — APPROVED** [review]
I'm happy with this. I'll let the maintainers review it further.

**Review by viceice — APPROVED** [review]

**Pull request merged**